modin/data_management/functions/groupby_function.py

```diff
@@ -206,11 +242,37 @@ def reduce(
         groupby_args["level"] = list(range(len(df.index.names)))
 
         apply_func = cls.try_filter_dict(reduce_func, df)
-        result = apply_func(df.groupby(axis=axis, **groupby_args), **reduce_args)
+        grp = df.groupby(axis=axis, **groupby_args)
+        if partition_selection is not None and not isinstance(reduce_func, dict):
+            # Some of the selected columns could've been dropped in the Map phase as non-suitable
```

Comment:
Can you provide an example for this case?

Reply:
```python
>>> df = pandas.DataFrame({"a": [1, 1, 2], "b": [3, 4, 5], "c": [5, 4, 3], "str_col": ["a", "b", "c"]})
>>> df.groupby("a")[["b", "c", "str_col"]].sum()
   b  c
a
1  7  9
2  5  3
>>>
```
Although I selected "str_col" for an aggregation it's not presented in the resulted frame, so if at the map phase `partition_selection=["b", "c", "str_col"]` it's not guaranteed that at the reduce phase `partition_selection` can be used without any modifications

Reply:
I see, thanks!